editor/components/colors/with-colors.js

```diff
@@ -14,6 +14,40 @@ import { withSelect } from '@wordpress/data';
  */
 import { getColorValue, getColorClass, setColorValue } from './utils';
 
+const memoizedGetColor = memoize(
+	( colors ) =>
+		memoize(
+			( colorName, customColorValue, colorContext ) => {
+				return {
+					name: colorName,
+					class: getColorClass( colorContext, colorName ),
+					value: getColorValue( colors, colorName, customColorValue ),
+				};
+			},
+			( colorName, customColorValue, colorContext ) =>
+				`${ colorName }-${ customColorValue }-${ colorContext }`
+		)
+);
+/**
+ * Even though, we don't expect setAttributes to change memoizing it is essential.
+ * If it is not memoized, each time memoizedSetColor is called,
+ * even if setAttributes was the same as before a new function reference is returned.
```

Comment:
Nitpick here but this is hard to parse. Could you modify the sentence a bit so it's easier to follow cause-and-effect, eg:

```js
* If setAttributes is not memoized, each time memoizedSetColor is called:
* a new function reference is returned (even if setAttributes has not changed).
```

Reply:
Thank you for the improvements they were applied 👍